### PR TITLE
bench: add bench script

### PR DESCRIPTION
**Pull request by GanZiheng**
Signed-off-by: GanZiheng [ganziheng98@gmail.com](mailto:ganziheng98@gmail.com)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,9 +34,9 @@ tikv-jemallocator = "0.4.0"
 
 [workspace]
 members = [
+    "agate_bench",
     "proto",
-    "skiplist",
-]
+    "skiplist"]
```
Comment (skyzh) [inline Cargo.toml:39]: new line?

File: agate_bench/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "agate_bench"
+version = "0.1.0"
+authors = ["Alex Chi <iskyzh@gmail.com>"]
```
Comment (skyzh) [inline agate_bench/Cargo.toml:4]: change to TiKV authors?

**Review by skyzh — APPROVED** [review]

**Review by Connor1996 — COMMENTED** [review]
do you plan to integrate continuing bench in this PR?

File: agate_bench/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,518 @@
...
+                            total.now()
+                        );
+                    }
+                }
+            }
+            let now = std::time::Instant::now();
+            println!(
+                "read total {} keys in {}",
+                total.now(),
+                now.duration_since(begin).as_secs_f64()
```
Comment (Connor1996) [inline agate_bench/src/main.rs:513]: `begin.elapsed()`, same for others

**Comment by GanZiheng** [comment]
> do you plan to integrate continuing bench in this PR?

No, I think this PR is enough. I will add more benches in https://github.com/tikv/agatedb/pull/169.

File: agate_bench/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+            let key_nums: u64 = sub_matches.value_of("key_nums").unwrap().parse().unwrap();
+            let value_size: usize = sub_matches.value_of("value_size").unwrap().parse().unwrap();
+            let chunk_size: u64 = sub_matches.value_of("chunk_size").unwrap().parse().unwrap();
+            let mut opts = rocksdb::Options::default();
+            opts.create_if_missing(true);
+            opts.set_compression_type(rocksdb::DBCompressionType::None);
+
+            let db = Arc::new(rocksdb::DB::open(&opts, directory).unwrap());
+            let mut expected = 0;
+            // let pb = ProgressBar::new(key_nums);
```
Comment (Connor1996) [inline agate_bench/src/main.rs:341]: clean it

**Review by Connor1996 — APPROVED** [review]
LGTM

File: agate_bench/Cargo.toml [hunk 5]
```diff
@@ -0,0 +1,22 @@
+[package]
+name = "agate_bench"
```
Comment (Connor1996) [inline agate_bench/Cargo.toml:2]: Please ignore this path for codecov

File: .github/codecov.yml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+ignore:
```
Comment (Connor1996) [inline .github/codecov.yml:1]: you should update the `codecov.yml` in working dir, rather than `.github/codecov.yml`

Comment (GanZiheng) [reply .github/codecov.yml:1]: Let me take a shot.

**Pull request merged**